### DISCUSSION TITLE
Is there a way to cancel the XHR in version 7.

### DISCUSSION THREAD

**Discussion by mukitek** [katspaugh/wavesurfer.js#3297] [Q&A] [answered]
Calling destroy in version 7, doesn't cancel an existing xhr. Would it possible to do this or to add a method which does?

**Comment by katspaugh** [comment, marked as answer]
As [mentioned in the readme](https://github.com/katspaugh/wavesurfer.js#removed-methods), in v7, there's a new constructor option, `fetchParams`. You can use it to pass an [abort controller](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).

```js
const controller = new AbortController()
const signal = controller.signal

const wavesurfer = new WaveSurfer({
  container: '#...',
  url: 'https://...',
  fetchParams: { signal }
})

wavesurfer.on('destroy', () => {
  controller.abort()
})
```

**Reply by mukitek** [reply]
thanks for your help
